vanity/src/safe.rs

```diff
@@ -0,0 +1,180 @@
+//! Safe deployment implementation.
+
+use crate::{create2::Create2, miner::Deployment, settings::SafeParameters};
+use ethereum_types::{Address, H256, U256};
+use hex_literal::hex;
+use rand::Rng;
+use tiny_keccak::{Hasher as _, Keccak};
+
+const PROXY_FACTORY: Address = address!("a6B71E26C5e0845f74c812102Ca7114b6a896AB2");
+const PROXY_INIT_CODE: H256 =
+    digest!("56e3081a3d1bb38ed4eed1a39f7729c3cc77c7825794c15bbf326f3047fd779c");
+const FALLBACK_HANDLER: Address = address!("f48f2B2d2a534e402487b3ee7C18c33Aec0Fe5e4");
+
+/// Safe deployment.
+#[derive(Clone)]
+pub struct Safe {
+    owners: Vec<Address>,
+    threshold: usize,
+    salt: [u8; 64],
+    create2: Create2,
+}
+
+impl Safe {
+    /// Creates a new safe from deployment parameters.
+    pub fn new(parameters: SafeParameters) -> Self {
+        // Safe requires sorted owners.
+        let mut owners = parameters.owners;
+        owners.sort_unstable();
```

Comment:
I don't think this is true for creating a safe: https://github.com/gnosis/safe-contracts/blob/main/contracts/base/OwnerManager.sol#L22-L43

It's true however for validating signature, it's done to make it cheaper to tell that there are no repeating addresses.

Reply:
Oops! 🙈

Reply:
Removed!